python/pyarrow/io.pyx

```diff
@@ -37,6 +37,10 @@ import sys
 import threading
 import time
 
+# To let us get a PyObject* and avoid Cython auto-ref-counting
+cdef extern from "Python.h":
+    PyObject* PyBytes_Empty" PyBytes_FromStringAndSize"(
```

Comment:
A bit confusing that this is called `_Empty`. That's what it's used for here but it actually can do more.